url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/frontend/handlers.java

```diff
@@ -0,0 +1,959 @@
+                } finally
+                {
+                    try
+                    {
+                        out.close();
+                    } catch (IOException ex)
+                    {
+//                        FrontEnd.statuswindow.append("Error occured ...\n");
+//                        FrontEnd.statuswindow.append(ex.getStackTrace().toString());
+                         
+                        FrontEnd.appendToPane(FrontEnd.statuswindow,"Error Occured...\n",Color.BLACK);
+                         FrontEnd.appendToPane(FrontEnd.statuswindow,ex.getStackTrace().toString(),Color.BLACK);
+                        //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                        //frontend.FrontEnd.exceptionraised++;
+                    }
+                }
+//                FrontEnd.statuswindow.append("PROGRAM STARTED \n"); 
+                FrontEnd.appendToPane(FrontEnd.statuswindow,"PROGRAM STARTED\n",Color.BLACK);
+                //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                FrontEnd.appendToPane(FrontEnd.statuswindow,"Running..\n",Color.GREEN);
+                clean_branchtable();
+
+                clean_memtable();
+                Integer ct = new Integer(0);
+                ct = backend.FirstPass.scan(temp);
+                //System.out.print("ct returned: " + ct);
+//                FrontEnd.statuswindow.append("Run:\n");
+                //frontend.FrontEnd.statuswindow.setCaretPosition(frontend.FrontEnd.statuswindow.getText().length());
+                try
+                {
+                    backend.ScanFile ob = new backend.ScanFile(temp);
+                    update_branchtable(ct);
+                    update(handlers.regMode);
+
+                } catch (Exception e)
+                {
+                }
+                // Delete temp file when program exits.
+                temp.deleteOnExit();
+
+                // Write to temp file
+                //BufferedWriter out = new BufferedWriter(new FileWriter(temp));
+                //out.write("aString");
+                //out.close();
+            } catch (IOException e)
+            {
+            }
+        }
+
+      }
+   
+
+    static void assignIniFin(int a, int b)
+      {
+        ini_line = a;
+        fin_line = b;
+        //System.out.println(ini_line + ", " + fin_line + " in assignIniFin");
+      }
+   
+
+    void stepover()
+      {
+        frontend.FrontEnd.steprun.setVisible(true);
+        FrontEnd.stop_debug_mode.setVisible(true);
+        step_over.callEncounter = 0;
+        debugMode = 2;
+      }
+
+    void stepout()
+      {
+        frontend.FrontEnd.steprun.setVisible(true);
+        FrontEnd.stop_debug_mode.setVisible(true);
+        debugMode = 1;
+
+      }
+
+    void stepinto()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 90 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.